src/Auth0.ManagementApi/Models/UserBase.cs

```diff
@@ -102,5 +104,67 @@ public abstract class UserBase
         /// </summary>
         [JsonProperty("blocked")]
         public bool? Blocked { get; set; }
+
+        /// <summary>
+        /// Returns <see cref="AppMetadata"/> as specific type.
+        /// </summary>
+        /// <typeparam name="T">Type to be returned.</typeparam>
+        /// <returns>An instance of T.</returns>
+        /// <exception cref="ArgumentException"></exception>
+        public T GetAppMetadata<T>()
+        {
+            if (AppMetadata is null)
+            {
+                return default;
+            }
+
+            if (AppMetadata is JObject jObject)
+            {
+                return jObject.ToObject<T>();
+            }
+
+            throw new ArgumentException("Unknown metadata type");
```

Comment:
I would just return null here and drop the first part that returns default. Default can be useful when we support more than just objects, but given it has to be an object, I guess we can use null for simplicity.

Having said that, Should we ensure T is an object?